test-plan/eventing/channel-lifecycle-conformance.md

```diff
@@ -0,0 +1,229 @@
+
+>> When a Channel is created, its spec.channelTemplate field MAY be populated to indicate which of several possible Channel implementations to use. It is RECOMMENDED to default the spec.channelTemplate field on creation if it is unpopulated. Once created, the spec.channelTemplate field MUST be immutable; the Channel MUST be deleted and re-created to change the spec.channelTemplate. This pattern is chosen to make it clear that changing spec.channelTemplate is not an atomic operation and that any implementation would be likely to result in event loss during the transition.
+
+
+
+# Testing Channel Lifecycle Conformance: 
+
+We are going to be testing the previous paragraphs coming from the Knative Eventing Spec. To do this we will be creating a Channel, checking its immutable properties, checking its Ready status and then creating a Subscription that links to it by making a reference. We will also checking the Subscription status, as it depends on the Channel to be ready to work correctly. We will be also checking that the Channel is addressable by looking at the status conditions fields. Because this is a Control Plane test, we are not going to be sending Events to the components. 
+
+You can find the resources for running these tests inside the [control-plane/channel-lifecycle/](channel-lifecycle/) directory. 
+- A [Channel resource](channel-lifecycle/channel.yaml)
+- A [Subscription resource that references the Channel](channel-lifecycle/subscription.yaml)
+- A [Services resources that serves as deadletter sink and subscriber for the Subscritpion](channel-lifecycle/services.yaml)
+- A [PingSources resources that sends events to the Channel](channel-lifecycle/ping-sources.yaml)
+
+
+## [Pre] Creating a Channel 
+
+```
+kubectl apply -f control-plane/channel-lifecycle/channel.yaml
+```
+
+
+## [Test] Immutability
+
+Check for default annotations, this should return the name of the selected implementation: 
+
+```
+kubectl get channel.messaging.knative.dev conformance-channel -o jsonpath='{.spec.channelTemplate.kind}'
+```
+
+Try to patch the annotation: `messaging.knative.dev/channel.spec.channelTemplate` to see if the resource mutates: 
+
+```
+kubectl patch channel.messaging.knative.dev conformance-channel --type merge -p '{"spec":{"channelTemplate":{"kind":"mutable"}}}'
+```
+
+You should get the following error: 
+```
+Error from server (BadRequest): admission webhook "validation.webhook.eventing.knative.dev" denied the request: validation failed: Immutable fields changed (-old +new): spec
+{v1.ChannelSpec}.ChannelTemplate.TypeMeta.Kind:
+	-: "InMemoryChannel" // or your channel implementation
+	+: "mutable"
+```
+
+Tested in eventing:
+- Not tested (depends on the implementation)
```

Comment:
This is ok, but I get a feeling we could be more explicit for the reader that just want to execute manual tests. My line is also not great, do you have a better suggestion?

```suggestion
Automated conformance test in Knative Eventing:
- Not tested (depends on the implementation)
```